code/object/objectdock.cpp

```diff
@@ -733,6 +831,30 @@ void object_remove_arriving_stage2_ndl_flag_helper(object *objp, dock_function_i
 		Ships[objp->instance].flags.remove(Ship::Ship_Flags::Arriving_stage_2_dock_follower);
 }
 
+void dock_calc_total_moi_helper(object* objp, dock_function_info* infop)
+{
+	matrix local_moi, unorient, temp, world_moi;
+	vm_inverse_matrix(&local_moi, &objp->phys_info.I_body_inv);
+	vm_copy_transpose(&unorient, &objp->orient);
+	vm_matrix_x_matrix(&temp, &objp->orient , &local_moi);
+	vm_matrix_x_matrix(&world_moi, &temp, &unorient);
```

Comment:
it would be a good idea to add comments and write out what you're doing here, using mathematical formulas, for the benefit of future programmers who look at the code